test/hooks.test.js

```diff
@@ -2716,7 +2716,7 @@ test('preSerialization hook should run before serialization and be able to modif
   })
 })
 
-test('preSerialization hook should be able to throw errors which are not validated against schema response', t => {
+test('preSerialization hook should be able to throw errors which are not validated against schema response', { skip: 'is this wanted?' }, t => {
```

Comment:
This test assumes that when an `Error` is thrown, the HTTP 500 schema defined by the user is ignored and fastify use its internal default.

I'm not sure how to handle it right now, In my opinion, an HTTP 500 status code could not have 2 different output:
- one defined by the user
- one when "something goes really wrong"

Does it make sense to you @fastify/core ?

Reply:
I'm not sure this is wanted. I'd recommend to check with blame when this was added and if there was a key reason to do this. At first glance, it does not seem correct.

Reply:
That test has been introduced with the `preSerialization` hook by https://github.com/fastify/fastify/pull/1408 
It was not intended to fix an open bug tho.

The use case it covers would be:
the user customizes its 5xx schemas and when the response is being serialized the user set a `next` error that would be different from the schema previously set

I would fix the test, checking that the schema set by the user would be applied if we agree that it would not be a breaking change

Reply:
I agree that this needs fixing. I would land this in a minor but it can also go in the next major.

Reply:
Done